### ISSUE TITLE
windows generate process fail

### ISSUE THREAD

**Issue by frankLife** [ericvicenti/ssh-keygen#10]
ssh-keygen is a good tool to make a ssh key easily. But when I use it on window 7 system. It would fail. the callback will not return error object at the same time. 

I look up the source code and find the bin which the program use . I click it directly. It throw a error like this

![ssh-keygen-error](https://user-images.githubusercontent.com/5081884/31503189-9bef6296-afa1-11e7-97f5-3d022176793f.PNG)

and I change the bin with [portable msysgit bin](https://github.com/msysgit/msysgit/releases) [portalble link](https://github.com/msysgit/msysgit/releases/download/Git-1.9.5-preview20150319/PortableGit-1.9.5-preview20150319.7z) using the files like

![ssh-keygen-bin](https://user-images.githubusercontent.com/5081884/31503516-7079b066-afa2-11e7-8ac8-f55fca311dd9.PNG)

It work again.

I hope the problem can be solved to make the project better

**Comment by ericvicenti** [comment]
If anybody wants to debug this and submit a pull request to fix it, I would appreciate it!

**Comment by jakkaj** [comment]
Hey - I've picked this one up. I have it working on 64bit machine with ssh-keygen from the latest portable Git for Windows. Just gigging out a 32-bit vm to test the other!

Will PR soon.

**Comment by mazensharkawy** [comment]
SOLVED !!!
After a bit of debugging. There is an exe file in the bin directory of this library. When you try to open any of them manually you get an error saying that there are missing dll files related to MSYS. It took me all day to install MSYS properly. so here are the links to solve this problem as there are a lot of sources out there that didn't solve the problem.

1. https://www.youtube.com/watch?v=ySE4HN4jcVw (follow the instructions in this tutorial)
2. if you have a missing dll still ... download this dll helper and it will fix it for you https://www.az-partners.net/apps/dll-helper/download?ap=25
**Bottom line is the problem is with the MYSYS and MINGW on your pc and not with this library**

**Comment by jakkaj** [comment]
Hey - I popped a PR in for this a while ago; https://github.com/ericvicenti/ssh-keygen/pull/13. If you pull that branch it should work...

**Comment by jeremyfix** [comment]
I faced this exact same issue indeed due to missing msys dlls . 

Actually I have also seen that on some windows machine (do not know if this is the case for all of them now),  ssh-keygen can be executed from the windows terminal directly. But I do not know how far this is true;

So I will probably try to use the PR #13 .
